### ISSUE TITLE
Text answer visibility info

### ISSUE THREAD

**Issue by janno42** [fsr-itse/EvaP#1133]
Next to each text question on the student's voting page a tooltip info icon should show which users (listed by their names) will be able to see the text result after the course was published. This includes the person to which the text answer belongs (+delegates), the responsible person (+delegates) and all other contributors who can see all text answers. A final line will state something like "and the staff users/evaluation team/student representatives" or similar (without listing individual names).

**Comment by karyon** [comment]
i have a bit of the fear that this list is by default rather long, might induce the feeling to voters that their comments are not confidential at all, and might keep them back from speaking freely.

just an idea, maybe a one-time email to responsibles/editors that we now display this information, they can use the preview to see this and this might be a good time to check whether the list of delegates is reasonably short, and they shouldn't be too liberal in giving viewing rights per course.

But maybe this isn't actually a problem in practice...

**Comment by janno42** [comment]
you're right. when writing the issue i felt it will not improve the situation. so before anyone implements this we need to discuss it in more detail.

**Comment by karyon** [comment]
for the record i still think this is a good thing since it adds transparency and users can make more informed decisions. however, if the result in practice is less open feedback, we might need other/additional solutions.

**Comment by janno42** [comment]
in the end this feature should help students to understand what's happening with their answers. they should know who will possibly work with the results and where the feedback will end up.

**Comment by karyon** [comment]
one option would be to just write the number of delegates instead of their names to make the list less intimidating. this might be a good idea anyway because we might not want to leak the list of delegates of all the contributors to all the participants.

with this, the average tooltip for a contributor (assuming they have no delegates) could be something like this:

"Besides the evaluated person themselves, this comment can be seen by Prof. Dr. John Smith, their 2 delegates, and the student representatives."

I think that's okay.

**Comment by chrisma** [comment]
@janno42 Thanks for pointing me to this ticket!

This issue seems like an important and valuable improvement to the system and its transparency to me.
For example, I intuitively assumed that the current field "Kommentare zur Person / Personal Feedback" would be viewable only by the person in question and not by responsible persons (+delegates) .

**Comment by karyon** [comment]
@janno42 i think we should think about making the `personal feedback`-question private by default (which would require implementation work) because of what @chrisma said.

also, let's think about sending an email to the responsibles that they should check their list of delegates when this goes live. or maybe the FSR could do that manually.

**Comment by karyon** [comment]
@chrisma i'd like to thank you for your comments, they led to https://github.com/fsr-itse/EvaP/pull/1279 which restricts the visibility of comments given on contributor's questions. they are still visible to the contributors themselves, their delegates and FSR members, but not the course's responsible anymore.

**Comment by chrisma** [comment]
Thanks for your work on this issue and moving EvaP forward!

[Originally](https://github.com/fsr-itse/EvaP/issues/1133#issue-306648646), this issue was about:
> Next to each text question on the student's voting page a tooltip info icon should show which users (listed by their names) will be able to see the text result after the course was published

I still consider this display of who will be able to see which answers after an evaluation is published (as a tooltip or some other form) an important part of increasing the transparency of the system. The level of detail / the specifics of what course participants are willing to share on EvaP will most likely also depend on their assumptions on who will read those comments. If the assumptions of visibility do not match the implementation, this might lead to some confusion.

As far as I can see #1279 does not change the way that information on visibility is displayed to users, but simply changes the default visibility of personal feedback. I'm not able to comment on whether this behavior is better or worse than before. While the course responsible person might not get to read "nasty" personal comments regarding their HiWis and WiMis, they might also miss personal praise. This seems like a design decision to me that could be argued either way.

#1279 seems to change the implementation details, but does not change the display of information on answer visibility.

I think @janno42's original comment is still very relevant:
> in the end this feature should help students to understand what's happening with their answers. they should know who will possibly work with the results and where the feedback will end up.

**Comment by karyon** [comment]
we fixed the original issue as well, see https://github.com/fsr-itse/EvaP/pull/1258. when a student evaluates a course, it looks like this:
![image](https://user-images.githubusercontent.com/1891915/48152287-bea84900-e2c3-11e8-87d2-0f3f92183d36.png)

in practice, contributors never had any text question other than the "personal feedback" question. so in practice, #1279 changed nothing more than making answers to that questions private.

regarding other potential text questions for contributors, you are right this is a design decision that can be argued either way. we opted for more data privacy and not automatically showing personal criticism (and praise, yes) to the bosses of contributors :) actually, that decision has been made by the FSR and has been confirmed by the dean of studies, so we/they are probably not going to change that in the near future.

**Comment by chrisma** [comment]
> we fixed the original issue as well, see #1258. when a student evaluates a course
Nice! Thanks for the pointer, that's definitely an improvement to the system 👍 

>  that decision has been made by the FSR and has been confirmed by the dean of studies, so we/they are probably not going to change that in the near future.
Good to know, thanks for the info!

**Comment by janno42** [comment]
Thanks also from my side about the discussion that lead to this. I do also think that it improves or handling of the data quite a lot.
The second part of your conversation about other text answers to contributors has also been thoroughly discussed and we're quite sure about the decision we made. We trust responsible and other contributors to discuss evaluation results in what manner they find to be appropriate and don't fear that praise will get lost. Incentives for personal discussions are still be given by the voting results and the continued review by the evaluation team and the dean of studies will make sure that critical problems won't be missed.
